internal/controllers/operator_controller.go

```diff
@@ -124,8 +132,19 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 		setResolvedStatusConditionUnknown(&op.Status.Conditions, "validation has not been attempted as spec is invalid", op.GetGeneration())
 		return ctrl.Result{}, nil
 	}
+
+	// gather variables for resolution
+	variables, err := r.variables(ctx)
```

Comment:
Ugh... multiple things named `variables`.

Reply:
Does `r.variables` need to be a function? Could we just do everything in that function here?

Reply:
I want to keep the function. If we do everythign here - I think readability will be decresed significantly (more stuff in `reconcile`) and we will have to handle each `if err != nil` branch separately (set conditions, etc).

I'm happy to rename `variables` into `vars` as Todd suggested. Will update the PR shortly.